python/adelphi/adelphi/nb.py

```diff
@@ -113,15 +150,28 @@ def __build_main_write_statement(self):
         return {"tags":{"name":"main-insert"}, "main-insert": build_insert_statements(self.keyspace, self.table)}
 
 
+    def __build_bindings(self, table):
+        (pk_cols, plain_cols) = partition_cols(table)
+        rv = {dist_binding_name(plain_col): self.__get_dist(plain_col.cql_type) for plain_col in plain_cols if is_supported_type(plain_col)}
+
+        def pk_generator():
+            for pk_col in pk_cols:
+                if not is_supported_type(pk_col):
+                    raise UnsupportedPrimaryKeyTypeException("No sequence definition for primary key column {} of type {}".format(pk_col.name, pk_col.cql_type))
+                yield((seq_binding_name(pk_col), self.__get_seq(pk_col.cql_type)))
+                # Each pk col also gets a DIST def because we may want to execute selects against it
+                yield((dist_binding_name(pk_col), self.__get_dist(pk_col.cql_type)))
+        rv.update(dict(pk_generator()))
+        return rv
+
+
     def __build_schema(self):
         """Really more of a config than a schema, but we'll allow it"""
         root = {}
 
-        rampup_scenario = "run driver=cql tags==phase:rampup cycles===TEMPLATE(rampup-cycles,1000) threads=auto"
-        main_scenario = "run driver=cql tags==phase:main cycles===TEMPLATE(main-cycles,1000) threads=auto"
-        root["scenarios"] = {"default":[rampup_scenario, main_scenario]}
+        root["scenarios"] = {"default":[self.__get_rampup_scenario(), self.__get_main_scenario()]}
```

Comment:
The scenario name should be templated too, we use different scenario names to distinguish between the workload executions in Grafana (despite using the same workload YAML). We set the names at runtime as a command-line arg.
Take a look at values.yaml, the current sample schema has a template tag around the scenario name.

Reply:
Ah, you're quite right, I did miss that initially.  Will fix shortly.

Reply:
This should be addressed now.

Reply:
I thought about this some over the weekend and I'm not sure this is the right solution in the end.  It seems like we should make this consistent with the cycles args discussed above; we should either support parameterization or not but I'm not sure there's a clear case for some things being specified via "adelphi" args and some via nosqlbench args.  I'm gonna take a run at converting this to following the pattern used for the cycles elsewhere.

Reply:
After pondering this a bit more I'm going to revert this work and go back to the idea of a scenario name as a nosqlbench param.  The impl in place now isn't really right; we'd need to run the generation process twice, once with a scenario name of "source" and once with a name of "target", simply to generate two configs which are otherwise identical.  This seems like overkill for a label which doesn't impact test functioning in any meaningful way.  We do need to provide a default value so that _some_ label is available for nosqlbench runs but as long as that's in place we should be good.

Reply:
> The impl in place now isn't really right; we'd need to run the generation process twice, once with a scenario name of "source" and once with a name of "target", simply to generate two configs which are otherwise identical. 

That goes back to my original thought that we should specify whatever parameters through nosqlbench such that we don't have to regenerate the whole file. We just pick a default and override it at runtime if necessary.
Doing the same for the cycle args would be at least consistent.